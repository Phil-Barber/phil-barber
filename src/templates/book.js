import React from 'react';
import { graphql } from 'gatsby';
import { Review } from './review';

const BookTemplate = ({ data }) => {
  const post = data.markdownRemark;
  const { published, author, dateCompleted, rating } = post.frontmatter;
  const featuredImgFluid = post.frontmatter.featuredImage.childImageSharp.fluid;
  const details = [
    {
      attr: 'Published',
      value: published,
    },
    {
      attr: 'Author',
      value: author,
    },
    {
      attr: 'Read',
      value: dateCompleted,
    },
    {
      attr: 'Rating',
      value: rating,
    },
  ];

  return (
    <Review
      imageFluid={featuredImgFluid}
      details={details}
      title={post.frontmatter.title}
      description={post.frontmatter.excerpt}
    >
      <div dangerouslySetInnerHTML={{ __html: post.html }} />
    </Review>
  );
};

export default BookTemplate;

export const query = graphql`
  query BookQuery($slug: String!) {
    markdownRemark(fields: { slug: { eq: $slug } }) {
      html
      frontmatter {
        title
        published
        author
        dateCompleted
        rating
        featuredImage {
          childImageSharp {
            fluid(maxWidth: 800) {
              ...GatsbyImageSharpFluid
            }
          }
        }
      }
    }
  }
`;
