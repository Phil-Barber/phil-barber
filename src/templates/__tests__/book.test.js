import React from 'react';
import { shallow } from 'enzyme';
import { BookTemplate } from '../book';

describe('BookTemplate', () => {
  it('Matches snapshot', () => {
    const container = shallow(
      <BookTemplate
        data={{
          markdownRemark: {
            html: '<div>This is the book review</div>',
            frontmatter: {
              title: 'BookTemplate title',
              published: '2019',
              dateCompleted: '2018-02-11',
              author: 'Collin',
              rating: '4/5',
              featuredImage: {
                childImageSharp: {
                  fluid: 'ImageFluid',
                },
              },
            },
          },
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
