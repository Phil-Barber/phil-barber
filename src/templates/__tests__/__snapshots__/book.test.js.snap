// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookTemplate Matches snapshot 1`] = `
<Review
  details={
    Array [
      Object {
        "attr": "Published",
        "value": "2019",
      },
      Object {
        "attr": "Author",
        "value": "Collin",
      },
      Object {
        "attr": "Read",
        "value": "2018-02-11",
      },
      Object {
        "attr": "Rating",
        "value": "4/5",
      },
    ]
  }
  imageFluid="ImageFluid"
  title="BookTemplate title"
>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>This is the book review</div>",
      }
    }
  />
</Review>
`;
